### DISCUSSION TITLE
Is PolyNote still used in production @ Netflix?

### DISCUSSION THREAD

**Discussion by rockpunk** [polynote/polynote#1426] [Q&A] [answered]
I vaguely remember reading some comments somewhere in the codebase a long time ago that the polynote design was not recommended for production and it only started out as a hack so @jeremyrsmith could learn ZIO. (Of course, now I can't find this blurb, so perhaps I'm crazy! 😂 ). In the 4 years that polynote was opensourced, however, there's been significant advancements and products in the notebook space:

Old guard: Zepplin, jupyterlab/jupyterhub still around
Cloud-based sharing communities: colab, kaggle
New startups: hex.tech, paperspace, others?
Big Dawgs: Databricks, AWS Sagemaker still dominating the 'enterprise' notebook IDE market 

(However, I think you can still only share variables between python and scala cells in polynote!)

Given the above, I'm wondering if Netflix is continuing to maintain and use Polynote in production internally. If so, why is this still considered 'experimental'?  If not, what's the plan to finally get a stable 1.x release out?  Thanks kindly!

**Comment by jeremyrsmith** [comment]
> not recommended for production

I think what you're remembering is my "you probably shouldn't use it" stance on [uzhttp](https://github.com/polynote/uzhttp).

This was mainly to make sure people didn't start relying on that library. It basically does only what polynote needs and nothing else (but there wasn't a ZIO-based alternative at that time)

To answer your question – we do still use Polynote, and it still gets maintained (and even gets new features sometimes). We don't use it "in production"; I'm not a big fan of load-bearing notebooks in general and fortunately there hasn't been much demand for supporting that use case.

As for a "1.x" release... this is only my opinion – and apologies if I'm too candid – but personally I'd have such a high bar for "1.0" that I think it would require a big community to form (and organize and contribute) in order for that to happen. 1.x software has things like internationalization, accessibility, UX design and polish, established ecosystem/package manager type stuff, etc – things that aren't a huge concern for a project that basically works well enough to fulfill its function to its few maintainers.

**Comment by jonathanindig** [comment]
Second everything that @jeremyrsmith said. Just one thing - if by "is it used in production" you mean "are people using it every day to get work done" then the answer is yes, definitely, and there are no plans to change that. Like Jeremy said, we continue to support those users, maintain the repo and even add new features (as you can see from our PR/commit history where, e.g., just yesterday I spent way too long trying to fix the build)

**Comment by antonkulaga** [comment]
I see that you did not publish 0.6.1 despite 0.6.0 release having issues.  So I thought that project is almost abandoned and Netflix switched to other notebook engines

**Reply by jonathanindig** [reply, marked as answer]
Polynote is still used at Netflix, as much as it's ever been, probably. 

Sorry we haven't released in a while. As you can see from the commit history, we are still actively maintaining the project. We release internally through a different process and thus haven't bothered to create a new tag here. We just deploy from the master branch. 

We were actually just talking about cutting a new official release soon.

**Reply by jeremyrsmith** [reply]
The blame is really with me. I changed laptops a while ago and the GPG key that I need to release to sonatype is still on the old laptop. I just haven't had time to deal with it.

That said, I don't blame @antonkulaga for thinking it's abandoned. It's true that we haven't worked on it that much. But it isn't because we don't use it, it's just because it's in a weird spot where it's not bad enough to need a lot of corporate-sponsored effort, but not good enough to get a lot of open-source community effort. Of course anyone can make a change to the latter!

**Comment by antonkulaga** [comment]
Well, my issue was that https://github.com/polynote/polynote/issues/1447 is not resolved. And if major issues (like broken release) are not resolved for multiple months it is a major sign of library being abandoned. In terms of releasing, if there are issues with antient systems like sonatype there are also easy to release things like https://jitpack.io
